src/ckan_to_bigquery.py

```diff
@@ -70,34 +81,84 @@ def search_sql(self, sql):
         '''
         NB: table must be full table id ...
         '''
-        # limit the number of results to ckan.datastore.search.rows_max
+        # limit the number of results to ckan.datastore.search.rows_max + 1
+        # (the +1 is so that we know if the results went over the limit or not)
         try:
              # check rows_max parameter set in CKAN config,
              # while testing as microlibrary (not as ckan ext) ckan is unknown
-            from ckan.common import config
             rows_max = int(config.get('ckan.datastore.search.rows_max', 32000))
         except:
-            rows_max = 32000 # set default rows limit 
-        
+            rows_max = 32000 # set default rows limit  
+        sql_initial = sql
         # limit the number of results to return by rows_max
-        sql = 'SELECT * FROM ({0}) AS blah LIMIT {1} ;'.format(sql, rows_max)
-        query = sql
-        query_job = self.bqclient.query(query)
-        rows = query_job.result()
+        sql = 'SELECT * FROM ({0}) AS blah LIMIT {1} ;'.format(sql, rows_max+1)
+        query_job = self.bqclient_readonly.query(sql, job_config=self.job_config)
+        rows = query_job.result() 
         records = [dict(row) for row in rows]
-
-        # TODO: check if results truncated ...
-        # if results.rowcount == rows_max + 1:
-        #    data_dict['records_truncated'] = True
-
-        return {
+        # check if results truncated ...
+        if len(records) == rows_max + 1:
+            sql_query_job = self.bqclient.query(sql_initial)
+            # get temp table containing query result
+            destination_table = sql_query_job.destination
+            log.warning("destination table: {}".format(destination_table))
+            destination_urls = self.extract_table(destination_table, sql_initial)
+            log.warning("extract job result: {}".format(destination_urls))
```

Comment:
Why are we logging warnings? This should be info ...

Reply:
that just stopped working suddenly, there is nothing in logs with INFO level, only WARNING and ERROR.
I'll recheck it again.